### PR TITLE
Docker mirror

### PR DESCRIPTION
**Pull request by hunchback**
solve docker.io rate limit by using github registry to populate the docker cache (currently only for builds but later also for the registry serving k8s).

when wanting to use the cache (and bypass docker.io) then run:

```
make docker-mirror-read
```

### CODE REVIEW

**Comment by hunchback** [comment]
running the script yeilds:

```
aidan@AIDANSHRIBMAN-SDFUE6N:~/projects/the-mesh-for-data/hack/tools$ ./docker_mirror.sh
latest: Pulling from library/alpine
Digest: sha256:d9a7354e3845ea8466bb00b22224d9116b183e594527fb5b6c3d30bc01a20378
Status: Image is up to date for alpine:latest
docker.io/library/alpine:latest
SOURCE: docker pull quay.io/ibm/alpine:latest-alpine-x86_64:latest
MIRROR: docker pull docker.io/alpine:latest

master: Pulling from banzaicloud/bank-vaults
596ba82af5aa: Already exists
d9b45a05ad6c: Already exists
00f605ef9460: Already exists
244040a42bf8: Already exists
4c1619e9b347: Pull complete
835805a12ec3: Pull complete
106173ef7d5b: Pull complete
Digest: sha256:12b1bbcaad53ca39e33350571ab07f3582c93ce1008ff7ab35a9618e3333f6a2
Status: Downloaded newer image for banzaicloud/bank-vaults:master
docker.io/banzaicloud/bank-vaults:master
SOURCE: docker pull quay.io/ibm/banzaicloud/bank-vaults:master-bank-vaults-x86_64:master
MIRROR: docker pull docker.io/banzaicloud/bank-vaults:master

0.5.4: Pulling from kindest/kindnetd
Digest: sha256:b33085aafb18b652ce4b3b8c41dbf172dac8b62ffe016d26863f88e7f6bf1c98
Status: Image is up to date for kindest/kindnetd:0.5.4
docker.io/kindest/kindnetd:0.5.4
SOURCE: docker pull quay.io/ibm/kindest/kindnetd:0.5.4-kindnetd-x86_64:0.5.4
MIRROR: docker pull docker.io/kindest/kindnetd:0.5.4

v1.16.9: Pulling from kindest/node
Digest: sha256:7175872357bc85847ec4b1aba46ed1d12fa054c83ac7a8a11f5c268957fd5765
Status: Image is up to date for kindest/node:v1.16.9
docker.io/kindest/node:v1.16.9
SOURCE: docker pull quay.io/ibm/kindest/node:v1.16.9-node-x86_64:v1.16.9
MIRROR: docker pull docker.io/kindest/node:v1.16.9

1.3.1: Pulling from library/vault
Digest: sha256:2f797433dfc322d7ba6fa81074e010873c740eb45e83ff5ced49cba585f82a66
Status: Image is up to date for vault:1.3.1
docker.io/library/vault:1.3.1
SOURCE: docker pull quay.io/ibm/library/vault:1.3.1-vault-x86_64:1.3.1
MIRROR: docker pull docker.io/library/vault:1.3.1

8u131-jre-alpine: Pulling from library/openjdk
Digest: sha256:45e198ce1514c4716e5b747cc8de6f9d483694ff4be4fecac07c9f73ef79b99a
Status: Image is up to date for openjdk:8u131-jre-alpine
docker.io/library/openjdk:8u131-jre-alpine
SOURCE: docker pull quay.io/ibm/openjdk:8u131-jre-alpine-openjdk-x86_64:8u131-jre-alpine
MIRROR: docker pull docker.io/openjdk:8u131-jre-alpine

0.11: Pulling from openpolicyagent/kube-mgmt
e6b0cf9c0882: Already exists
74bdd6ab0c86: Pull complete
Digest: sha256:6ac4d979fda61e15123b43e8e6d2cd6163f3ae0f6dcf1785a8507012d827f8a3
Status: Downloaded newer image for openpolicyagent/kube-mgmt:0.11
docker.io/openpolicyagent/kube-mgmt:0.11
SOURCE: docker pull quay.io/ibm/openpolicyagent/kube-mgmt:0.11-kube-mgmt-x86_64:0.11
MIRROR: docker pull docker.io/openpolicyagent/kube-mgmt:0.11

latest: Pulling from openpolicyagent/opa
Digest: sha256:d5b2a82662ce18c3456645b73519592835c96e5493c0ebf35e7dcbc5474c730b
Status: Image is up to date for openpolicyagent/opa:latest
docker.io/openpolicyagent/opa:latest
SOURCE: docker pull quay.io/ibm/openpolicyagent/opa:latest-opa-x86_64:latest
MIRROR: docker pull docker.io/openpolicyagent/opa:latest

latest: Pulling from prom/statsd-exporter
Digest: sha256:d23aca343b869352b3391e506748ec741c1c22960fc02cef1c444e9ba083af56
Status: Image is up to date for prom/statsd-exporter:latest
docker.io/prom/statsd-exporter:latest
SOURCE: docker pull quay.io/ibm/prom/statsd-exporter:latest-statsd-exporter-x86_64:latest
MIRROR: docker pull docker.io/prom/statsd-exporter:latest

3.6: Pulling from library/python
Digest: sha256:da022140db3b40d07c81815158092ff8ccfd967926b533a7c0b573eeeb5be120
Status: Image is up to date for python:3.6
docker.io/library/python:3.6
SOURCE: docker pull quay.io/ibm/python:3.6-python-x86_64:3.6
MIRROR: docker pull docker.io/python:3.6

3.6-slim: Pulling from library/python
Digest: sha256:8b38a840436fbd61274d58e923270d519176ad19924744df27b1cfe3b8c5384b
Status: Image is up to date for python:3.6-slim
docker.io/library/python:3.6-slim
SOURCE: docker pull quay.io/ibm/python:3.6-slim-python-x86_64:3.6-slim
MIRROR: docker pull docker.io/python:3.6-slim

v0.0.12: Pulling from rancher/local-path-provisioner
Digest: sha256:53f839db5e2b28cce471ae599ea92bebaf3d7589ea8ee350992dc9f64486c7bc
Status: Image is up to date for rancher/local-path-provisioner:v0.0.12
docker.io/rancher/local-path-provisioner:v0.0.12
SOURCE: docker pull quay.io/ibm/rancher/local-path-provisioner:v0.0.12-local-path-provisioner-x86_64:v0.0.12
MIRROR: docker pull docker.io/rancher/local-path-provisioner:v0.0.12

2: Pulling from library/registry
Digest: sha256:d5459fcb27aecc752520df4b492b08358a1912fcdfa454f7d2101d4b09991daa
Status: Image is up to date for registry:2
docker.io/library/registry:2
SOURCE: docker pull quay.io/ibm/registry:2-registry-x86_64:2
MIRROR: docker pull docker.io/registry:2

1.3.1: Pulling from library/vault
Digest: sha256:2f797433dfc322d7ba6fa81074e010873c740eb45e83ff5ced49cba585f82a66
Status: Image is up to date for vault:1.3.1
docker.io/library/vault:1.3.1
SOURCE: docker pull quay.io/ibm/vault:1.3.1-vault-x86_64:1.3.1
MIRROR: docker pull docker.io/vault:1.3.1
```

**Comment by hunchback** [comment]
output when running form `make`:

```
aidan@AIDANSHRIBMAN-SDFUE6N:~/projects/the-mesh-for-data$ make docker-mirror-read
./hack/tools/docker_mirror.sh ./hack/tools/docker_mirror.conf
latest: Pulling from ibm/alpine-x86_64
Digest: sha256:d9a7354e3845ea8466bb00b22224d9116b183e594527fb5b6c3d30bc01a20378
Status: Image is up to date for quay.io/ibm/alpine-x86_64:latest
quay.io/ibm/alpine-x86_64:latest
$ docker pull quay.io/ibm/alpine-x86_64:latest
$ docker tag quay.io/ibm/alpine-x86_64:latest docker.io/alpine:latest

latest: Pulling from ibm/bank-vaults-x86_64
Digest: sha256:65d58abbd18ba8773a0b1fc764b771bd48d954944f245477e58aa50d6d6d8e3d
Status: Image is up to date for quay.io/ibm/bank-vaults-x86_64:latest
quay.io/ibm/bank-vaults-x86_64:latest
$ docker pull quay.io/ibm/bank-vaults-x86_64:latest
$ docker tag quay.io/ibm/bank-vaults-x86_64:latest docker.io/docker.io/banzaicloud/bank-vaults:master

0.5.4: Pulling from ibm/kindnetd-x86_64
Digest: sha256:b33085aafb18b652ce4b3b8c41dbf172dac8b62ffe016d26863f88e7f6bf1c98
Status: Image is up to date for quay.io/ibm/kindnetd-x86_64:0.5.4
quay.io/ibm/kindnetd-x86_64:0.5.4
$ docker pull quay.io/ibm/kindnetd-x86_64:0.5.4
$ docker tag quay.io/ibm/kindnetd-x86_64:0.5.4 docker.io/kindest/kindnetd:0.5.4

1.3.1: Pulling from ibm/vault-x86_64
Digest: sha256:2f797433dfc322d7ba6fa81074e010873c740eb45e83ff5ced49cba585f82a66
Status: Image is up to date for quay.io/ibm/vault-x86_64:1.3.1
quay.io/ibm/vault-x86_64:1.3.1
$ docker pull quay.io/ibm/vault-x86_64:1.3.1
$ docker tag quay.io/ibm/vault-x86_64:1.3.1 docker.io/library/vault:1.3.1

8u131-jre-alpine: Pulling from ibm/openjdk-x86_64
Digest: sha256:45e198ce1514c4716e5b747cc8de6f9d483694ff4be4fecac07c9f73ef79b99a
Status: Image is up to date for quay.io/ibm/openjdk-x86_64:8u131-jre-alpine
quay.io/ibm/openjdk-x86_64:8u131-jre-alpine
$ docker pull quay.io/ibm/openjdk-x86_64:8u131-jre-alpine
$ docker tag quay.io/ibm/openjdk-x86_64:8u131-jre-alpine docker.io/openjdk:8u131-jre-alpine

3.6: Pulling from ibm/python-x86_64
Digest: sha256:da022140db3b40d07c81815158092ff8ccfd967926b533a7c0b573eeeb5be120
Status: Image is up to date for quay.io/ibm/python-x86_64:3.6
quay.io/ibm/python-x86_64:3.6
$ docker pull quay.io/ibm/python-x86_64:3.6
$ docker tag quay.io/ibm/python-x86_64:3.6 docker.io/python:3.6

3.6-slim: Pulling from ibm/python-x86_64
Digest: sha256:8b38a840436fbd61274d58e923270d519176ad19924744df27b1cfe3b8c5384b
Status: Image is up to date for quay.io/ibm/python-x86_64:3.6-slim
quay.io/ibm/python-x86_64:3.6-slim
$ docker pull quay.io/ibm/python-x86_64:3.6-slim
$ docker tag quay.io/ibm/python-x86_64:3.6-slim docker.io/python:3.6-slim

1.3.1: Pulling from ibm/vault-x86_64
Digest: sha256:2f797433dfc322d7ba6fa81074e010873c740eb45e83ff5ced49cba585f82a66
Status: Image is up to date for quay.io/ibm/vault-x86_64:1.3.1
quay.io/ibm/vault-x86_64:1.3.1
$ docker pull quay.io/ibm/vault-x86_64:1.3.1
$ docker tag quay.io/ibm/vault-x86_64:1.3.1 docker.io/vault:1.3.1

aidan@AIDANSHRIBMAN-SDFUE6N:~/projects/the-mesh-for-data$ git diff
```

File: Makefile [hunk 1]
```diff
@@ -123,6 +127,8 @@ define do-docker-retag-and-push-public
 endef
 
 .PHONY: docker-retag-and-push-public
+docker-retag-and-push-public: export DOCKER_HOSTNAME?=kind-registry:5000
```
Comment (roee88) [inline Makefile:130]: Why is this changed? Shouldn't it stay as the public vars?

Comment (hunchback) [reply Makefile:130]: removed it from the .travis env to makefile (thats' all) so that when running the command from the makefile it runs as it should

**Review by roee88 — APPROVED** [review]
